include/gridtools/common/defs.hpp

```diff
@@ -26,6 +27,21 @@ namespace gridtools {
 #endif
 #endif
 
+#if defined(__CUDACC__) &&                                                                       \
+    (__CUDACC_VER_MAJOR__ >= 12 || (__CUDACC_VER_MAJOR__ == 11 && __CUDACC_VER_MINOR__ >= 2)) && \
+    defined(__CUDA_ARCH__)
```

Comment:
Maybe this is enough?
```suggestion
#if defined(__has_builtin) && __has_builtin(__builtin_assume)
```

Reply:
I like it.